### ISSUE TITLE
Allways return checksums for all files even if empty, add a new filetype flag `is_empty`

### ISSUE THREAD

**Issue by pombredanne** [nexB/scancode-toolkit#461]
Following #459 we need to always return checksums such as SHA1 and add a new `is_empty` flag for empty files.

**Comment by sschuberth** [comment]
While working on this, it probably makes sense to consider adding more flags that also are orthogonal to the current flags like `is_link`, `is_file` etc. I'm mainly thinking of `is_binary` (and / or `is_source`) and `is_archive` to have direct mappings to the SPDX `FileType` field.

**Comment by kartiksibal** [comment]
@pombredanne Since, you have explicitly mentioned in #457 the reason for not using a hash for an empty file. Could you please elucidate your expectations with this?

Thanks :-)

**Comment by pombredanne** [comment]
are you sure this was #457 ? that does not sounds related?

**Comment by kartiksibal** [comment]
@pombredanne I might have interpreted it wrong. I had based my conclusion on, 

![screenshot from 2017-03-11 00-48-47](https://cloud.githubusercontent.com/assets/10362901/23809736/dce54cc2-05f4-11e7-865d-115b6b1fd959.png)

**Comment by arnav-mandal1234** [comment]
@pombredanne for this issue, I have to calculate sha1 and md5 for an empty file as well keep one extra boolean is_empty to keep track of empty files. Please guide me if I am wrong. :)

**Comment by pombredanne** [comment]
@arnav-mandal1234 correct, but I think this is no longer an issue. That would be worth checking on a --info scan that would have some empty files

**Comment by pombredanne** [comment]
Note that this is a possible issue in the SPDX output that demands to have a checksum even for an empty file.

>  So, should I assign a constant hash for the empty files? (all empty files would have same hash value)

That's likely the best approach

**Comment by sschuberth** [comment]
Note that at the example of SHA-1, the [hash for an empty string](https://en.wikipedia.org/wiki/SHA-1#Example_hashes) / no content is `da39a3ee5e6b4b0d3255bfef95601890afd80709`.

**Comment by pombredanne** [comment]
@sschuberth thanks! I think we should not return it always ... but rather only in SDPX outputs, because otherwise this has little value. Frankly SPDX-wise, demanding a checksum (and a combined document-level checksum) that includes empty files makes no sense to me. From a matching point of view, an empty file has no purpose either.

**Comment by pombredanne** [comment]
We have:
```
mime_type: inode/x-empty
file_type: empty
```
for empty files. I am still thinking that creating checksum empty files like done in SPDX is complete non-sense. But I will keep this open as a low priority item.
